File: src/ast/AST.java
package ast;

/*
 * Abstract Syntax Tree
 */
public class AST {
  private String text = "";
  private ASTNode root;

  public AST(ASTNode node) {
    this.root = node;
  }

  public String getAST() {
    preOrder(root, "");
    return text;
  }

  private void preOrder(ASTNode node, String printPrefix) {
    if (node == null)
      return;

    addNodeDetails(node, printPrefix);
    preOrder(node.getChild(), printPrefix + ".");
    preOrder(node.getSibling(), printPrefix);
  }

  private void addNodeDetails(ASTNode node, String printPrefix) {
    if (node.getType() == ASTNodeType.IDENTIFIER ||
        node.getType() == ASTNodeType.INTEGER)
      text += String.format(printPrefix + node.getType().getPrintName() + "\n", node.getValue());
    else if (node.getType() == ASTNodeType.STRING)
      text += String.format(printPrefix + node.getType().getPrintName() + "\n", node.getValue());
    else
      text += String.format(printPrefix + node.getType().getPrintName() + "\n");
  }
}

File: reCreate/main/AST/AST.py
from .ASTNode import nodeTypes

class AST:
    def __init__(self,root) -> None:
        self.text=""
        self.root=root
    def getAST(self):
       self.preOrder(self.root,"")
       return self.text
    def preOrder(self,node,printPrefix):
        if node==None:
            return
        self.addNodeDetails(node,printPrefix)
        self.preOrder(node.getChild(),printPrefix+".")
        self.preOrder(node.getSibling(),printPrefix)

    def addNodeDetails(self,node,printPrefix):
        # print("tempdddddd",node.getType())
        if node.getType()[0]==nodeTypes["IDENTIFIER"][0] or node.getType()[0]==nodeTypes["INTEGER"][0]:
           
            string=str(printPrefix+node.getPrintName()+"\n") % (node.getValue())
            self.text+=string
        elif node.getType()[0]==nodeTypes["STRING"][0]:
            string=str(printPrefix+node.getPrintName()+"\n") % (node.getValue())
            self.text+=string
        else:
            self.text+=printPrefix+node.getPrintName()+"\n"
